### PR TITLE
[Testing] Application testing in Symfony is fun! Let's document the modern features

### PR DESCRIPTION
**Pull request by wouterj**
This PR continues the efforts of restructuring the Testing docs:

- Focus more on Arrange, Act, Assert
- Advertise modern testing features (assertions). Somehow, we didn't mention any easy to use assertion in the main guide, but instead explained the complex dom crawler API.
- Reduce some "duplication". There is an astonishing number of ways to fill in and submit a form (we documented at least 4 different APIs in the main guide!). I've reduced this to two: a plain and short one, and a longer one for more complex form submissions.

Todo, preferably in a follow-up PR: combine the `testing/dom_crawler` and `components/dom_crawler` feature documentation into `testing/dom_crawler`.

cc @Nyholm

### CODE REVIEW

File: testing.rst [hunk 1]
```diff
@@ -732,167 +613,69 @@ to be reported by PHPUnit::
...
 
-Extracting Information
-......................
+On each request, you can enable the Symfony profiler to collect data about the
+internal handling of that request. For example, the profiler could be used to
+verify that a given page runs less than a certain number of database
+queries when loading.
 
-The Crawler can extract information from the nodes::
+To get the Profiler for the last request, do the following::
```
Comment (javiereguiluz) [inline testing.rst:650]: `Profiler` -> `profiler`

**Review by javiereguiluz — APPROVED** [review]
I like it! It's a nice simplification + reorganization. Thanks Wouter!

**Comment by javiereguiluz** [comment]
Thanks Wouter! This is now merged.

**Comment by wouterj** [comment]
Thank you @javiereguiluz. If you need help with the up merging, feel free to ping me :)

**Comment by javiereguiluz** [comment]
I've just finished upmerging. I think I did it right, but I'm not 100% sure, so maybe you could check the result. Thanks! I'll let you merge the other one (#15305) so you can easily spot the confilcts 🙏  Thanks!

**Pull request merged**